sdjwt/src/test/kotlin/web5/security/SdJwtBlinderTest.kt

```diff
@@ -0,0 +1,333 @@
+
+  private val mapper = jacksonObjectMapper().apply {
+    enable(SerializationFeature.INDENT_OUTPUT)
+    setSerializationInclusion(JsonInclude.Include.NON_NULL)
+    setDefaultPrettyPrinter(CustomPrettyPrinter())
+  }
+
+  @Test
+  fun testExample1() {
+    val claims = """{
+      "iss": "https://example.com/issuer",
+      "iat": 1683000000,
+      "exp": 1883000000,
+      "sub": "user_42",
+      
+      "cnf": {
+        "jwk": {
+          "kty": "EC",
+          "crv": "P-256",
+          "x": "TCAER19Zvu3OHF4j4W4vfSVoHIP1ILilDls7vCeGemc",
+          "y": "ZxjiWWbZMQGHVWKVQ4hbSIirsVfuecCE6t4jT9F2HZQ"
+        }
+      },
+      
+      "updated_at": 1570000000,
+      "email": "johndoe@example.com",
+      "phone_number": "+1-202-555-0101",
+      "family_name": "Doe",
+      "phone_number_verified": true,
+      "address": {
+        "street_address": "123 Main St",
+        "locality": "Anytown",
+        "region": "Anystate",
+        "country": "US"
+      },
+      "birthdate": "1940-01-01",
+      "given_name": "John",
+      "nationalities": [
+        "US",
+        "DE"
+      ]
+    }""".trimIndent()
+
+    val keyManager = InMemoryKeyManager()
+    DidKey.create(keyManager)
+
+    val signer = SdJwtBlinder(
+      saltGenerator = MockMapGenerator(
+        mapOf(
+          "given_name" to "2GLC42sKQveCfGfryNRN9w",
+          "family_name" to "eluV5Og3gSNII8EYnsxA_A",
+          "email" to "6Ij7tM-a5iVPGboS5tmvVA",
+          "phone_number" to "eI8ZWm9QnKPpNPeNenHdhQ",
+          "phone_number_verified" to "Qg_O64zqAxe412a108iroA",
+          "address" to "AJx-095VPrpTtN4QMOqROA",
+          "birthdate" to "Pc33JM2LchcU_lHggv_ufQ",
+          "updated_at" to "G02NSrQfjFXQ7Io09syajA",
+          "nationalities[0]" to "lklxF5jMYlGTPUovMNIvCA",
+          "nationalities[1]" to "nPuoQnkRFq3BIeAm7AnXFA",
+        )
+      ),
+      shuffle = {},
+      totalDigests = { i -> i },
+      mapper = mapper,
+    )
+    // Define claims to blind
+    // The nationalities array is always visible, but its contents are selectively disclosable.
+    // The sub element and essential verification data (iss, iat, cnf, etc.) are always visible.
+    // All other End-User claims are selectively disclosable.
+    // For address, the Issuer is using a flat structure, i.e., all of the claims in the address claim can only be disclosed in full. Other options are discussed in Section 5.7.
+    val claimsToBlind = mapOf(
+      "given_name" to FlatBlindOption,
+      "family_name" to FlatBlindOption,
+      "email" to FlatBlindOption,
+      "phone_number" to FlatBlindOption,
+      "phone_number_verified" to FlatBlindOption,
+      "address" to FlatBlindOption,
+      "birthdate" to FlatBlindOption,
+      "updated_at" to FlatBlindOption,
+      "nationalities" to ArrayBlindOption,
+    )
+
+    val sdJwt = signer.blind(claims, claimsToBlind)
+
+    val expected = """{
+      "_sd": [
+        "CrQe7S5kqBAHt-nMYXgc6bdt2SH5aTY1sU_M-PgkjPI",
```

Comment:
have you tested its the same output as ssi-sdk?

Reply:
The examples are coming from the spec itself, so we should be covered. In a later PR, I'll add loads of test vectors from the reference implementation.